### PR TITLE
[KG] Disable filter in evaluation

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
By default, we remove the positive edges from sampled negative edges in evaluation. We should allow disabling the filtering for fair comparison with other frameworks.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: apps/kg/eval.py [hunk 1]
```diff
@@ -99,17 +101,20 @@ def main(args):
     args.neg_sample_size_test = args.neg_sample_size
     if args.neg_sample_size < 0:
         args.neg_sample_size_test = args.neg_sample_size = eval_dataset.g.number_of_nodes()
+    args.eval_filter = not args.no_eval_filter
```
Comment (classicsong) [inline apps/kg/eval.py]: I think we cat directly change self.add_argument('--no_eval_filter' to --eval_filter

File: apps/kg/train.py [hunk 2]
```diff
@@ -59,6 +59,8 @@ def __init__(self):
                           help='margin value')
         self.add_argument('--eval_percent', type=float, default=1,
                           help='sample some percentage for evaluation.')
+        self.add_argument('--no_eval_filter', action='store_true',
```
Comment (classicsong) [inline apps/kg/train.py]: same here no_eval_filter -> eval_filter

File: apps/kg/eval.py (hunk 1 above)
Comment (zheng-da) [reply apps/kg/eval.py]: actually, i want to have filtering by default and disable filtering only when a user explicitly asks for it.

Comment (classicsong) [reply apps/kg/eval.py]: We can use: store_false

Comment (classicsong) [reply apps/kg/eval.py]: I see. This make sense

**Pull request merged**